tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,453 @@
+---
+title: Time, patterns and a little more
+published: 2021-11-25
+tags: haskell, pattern-synonyms, property-testing
+libraries: gtk3-0.14.4 glib-0.13.2.2
+language: haskell
+author: Felix Miño
+author-name: Felix Miño
+github-profile: felixminom
+description: In this tutorial we’re going to explore the use of pattern synonyms in Haskell, later we’ll give a brief view to the Stack Builders’ contribution to the `time` Haskell library and finally we’ll review how to use the utilities that were introduced in the contribution.
+---
+
+As Stack Builder one of the core values of our company is contributing
+to open source (OSS). We believe that we have the power to change
+people’s lives (or at least make it easier) by pushing the boundaries
+of the software industry and we’re happy to announce that the past month
+we were able to contribute to one of Haskell's core libraries, time.
+
+One of our tech leads, Sebastian Estrella, started with the idea:
+“What if we bring some utilities that Ruby on Rails (RoR) has to Haskell?”.
```

Comment:
Yes as I far as I know this was only available in Ruby on Rails, @sestrella  could you help us clarifying this?

Reply:
@felixminom yes, this is a feature only available in the context of Rails since by default Ruby does not provide these methods:

Rails (REPL)

```
➜  bundle exec rails c
Running via Spring preloader in process 77374
Loading development environment (Rails 6.1.4.1)
irb(main):001:0> Date.today.all_month
=> Mon, 01 Nov 2021..Tue, 30 Nov 2021
```

Ruby (REPL)

```
➜  irb
irb(main):001:0> require "date"
=> true
irb(main):002:0> Date.today.all_month
Traceback (most recent call last):
        4: from /nix/store/7nja5rmprz8sb72zvjbc1v85rswwsi4j-ruby-2.7.4/bin/irb:23:in `<main>'
        3: from /nix/store/7nja5rmprz8sb72zvjbc1v85rswwsi4j-ruby-2.7.4/bin/irb:23:in `load'
        2: from /nix/store/7nja5rmprz8sb72zvjbc1v85rswwsi4j-ruby-2.7.4/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):2
NoMethodError (undefined method `all_month' for #<Date: 2021-11-26 ((2459545j,0s,0n),+0s,2299161j)>)
irb(main):003:0>
```

Rails [monkey patch](https://en.wikipedia.org/wiki/Monkey_patch) some types in the std library